### PR TITLE
Add waitlist button functionality

### PR DESCRIPTION
**Pull request by mehuylsharma**
Related to: https://linear.app/opencap/issue/OPE-12/add-waitlist

image

### CODE REVIEW

File: src/app/api/waitlist/route.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { type NextRequest, NextResponse } from "next/server";
```
Comment (dephraiim) [inline src/app/api/waitlist/route.ts:1]: This should work well

```ts
import type { NextRequest, NextResponse } from "next/server";
```

File: src/app/page.tsx [hunk 2]
```diff
@@ -18,11 +37,60 @@ export default function HomePage() {
...
+              <Button
+                onClick={() => setModalOpen(true)}
+                className="hover:no-underline"
+              >
+                Join the waitlist
+              </Button>
+            </DialogTrigger>
+            <DialogContent className="sm:max-w-[425px]">
+              <form
+                onSubmit={async (event) => {
```
Comment (dephraiim) [inline src/app/page.tsx:51]: Can we go ahead with a library from the start? Like react-hook-form? The earlier the better

**Review by dephraiim — CHANGES_REQUESTED** [review]

File: src/app/page.tsx (hunk 2 above)
Comment (mehuylsharma) [reply src/app/page.tsx:51]: good suggestion, let me do that

**Comment by dahal** [comment]
> Latest build errors (a GitHub Action to share these automatically would be great lol):
> 
> image
> @dahal seeing lots of Prisma `any` warnings (eg. [this one](https://typescript-eslint.io/rules/no-unsafe-call/)) - thoughts on extending the Next.js TSConfig to get rid of these?

> Latest build errors (a GitHub Action to share these automatically would be great

Lets create an issue on this one ☝🏼 , perhaps someone from community can help.

> @dahal seeing lots of Prisma any warnings (eg. [this one](https://typescript-eslint.io/rules/no-unsafe-call/)) - thoughts on extending the Next.js TSConfig to get rid of these?

I'll look into this one.

File: src/app/api/waitlist/route.ts (hunk 1 above)
Comment (dephraiim) [reply src/app/api/waitlist/route.ts:1]: oh okay

**Review by tedspare — APPROVED** [review]
LGTM, great work 💪 🚢

**Comment by tedspare** [comment]
Ran `prisma migrate dev` to add a migration file and apply the schema changes. Then tested and worked in preview!

image

**Pull request merged**